### PR TITLE
Fix pagination bugs and refactor/abstract styling logic

### PR DESCRIPTION
**Pull request by delbaoliveira**
This PR fixes the pagination bugs:

1. We were using `pointer-events-none` to disable arrows for the first and last pages, but this only disables mouse events - keyboard users could navigate to `page=0` or `page=totalPages + 1`. This PR truly disables them by rendering a `<div>` instead of a `<Link>`. It also means we're not passing a href to disabled links.  

CleanShot 2023-10-09 at 13 32 56@2x

2. Fixes hover states. Removes hover state on active items, removes hover state on disabled items.
3. Disable ellipsis.  
4. Fixes border styles.

CleanShot 2023-10-09 at 14 03 47@2x

5. Fix bug where you could not click on subsequent pages if a search term existed (since the page would always default to 1)

I've also extracted away the logic into separate components to make it easier for the user to update the `<Pagination>` component.

### CODE REVIEW

File: dashboard/15-final/app/dashboard/customers/page.tsx [hunk 1]
```diff
@@ -4,12 +4,10 @@ import CustomersTable from '@/app/ui/customers/table';
...
 }: {
-  searchParams:
-    | {
-        query: string | undefined;
-        page: string | undefined;
-      }
-    | undefined;
+  searchParams?: {
+    query?: string;
+    page?: string;
```
Comment (delbaoliveira) [inline dashboard/15-final/app/dashboard/customers/page.tsx:9]: This is not related, just cleaning up TS.

**Review by emilkowalski — APPROVED** [review]
